lib/administrate/base_dashboard.rb

```diff
@@ -18,6 +18,18 @@ def attribute_types
       self.class::ATTRIBUTE_TYPES
     end
 
+    def attribute_type_for(attribute_name)
+      attribute_types.fetch(attribute_name) do
+        fail "Attribute #{attribute_name} could not be found in #{self.class}::ATTRIBUTE_TYPES"
```

Comment:
We can extract a method for the error message, similar to what you did in the tests. If the line's still too long after that, I'm ok ignoring it.